src/main/java/com/android/volley/toolbox/DiskBasedAsyncCacheFallback.java

```diff
@@ -0,0 +1,152 @@
+package com.android.volley.toolbox;
+
+import android.text.TextUtils;
+import androidx.annotation.VisibleForTesting;
+import com.android.volley.AsyncCache;
+import com.android.volley.VolleyLog;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * AsyncCache implementation that uses Java's standard IO libraries to perform disc reads and writes
+ * This should be used as a fallback for devices with an API level below 26, as the operations are
+ * synchronous.
+ */
+public class DiskBasedAsyncCacheFallback extends AsyncCache {
+
+    /** Map of the Key, CacheHeader pairs */
+    private final Map<String, CacheHeader> mEntries = new LinkedHashMap<>(16, .75f, true);
+
+    /** Total amount of space currently used by the cache in bytes. */
+    private long mTotalSize = 0;
+
+    /** The supplier for the root directory to use for the cache. */
+    private final DiskBasedCacheUtility.FileSupplier mRootDirectorySupplier;
+
+    /** The maximum size of the cache in bytes. */
+    private final int mMaxCacheSizeInBytes;
+
+    /** Default maximum disk usage in bytes. */
```

Comment:
Are these used anywhere yet? If not, I think hold off on including them until they're used.

When you do need to add them, I think you can put them in your new Utility class since they are the same for both implementations (async and fallback)

Reply:
I removed these for now